casper/src/test/scala/coop/rchain/casper/helper/TaskCasper.scala

```diff
@@ -0,0 +1,113 @@
+import coop.rchain.comm._
+import coop.rchain.crypto.signatures.Ed25519
+import coop.rchain.metrics.Metrics
+import coop.rchain.p2p.EffectsTestInstances._
+import coop.rchain.p2p.effects.PacketHandler
+import coop.rchain.comm.connect.Connect.dispatch
+import coop.rchain.comm.protocol.routing._
+import coop.rchain.rholang.interpreter.Runtime
+import java.nio.file.Files
+
+import coop.rchain.casper.util.rholang.RuntimeManager
+import monix.execution.Scheduler
+import monix.eval.Task
+import scala.concurrent.duration.{FiniteDuration, MILLISECONDS}
+import scala.collection.mutable
+import coop.rchain.shared.PathOps.RichPath
+import scala.util.Random
+
+object TaskCasper {
+  def apply(sk: Array[Byte], genesis: BlockMessage)(
+      implicit scheduler: Scheduler): (Task[MultiParentCasper[Task]], () => Unit) = {
+    val blockStoreDir = BlockStoreTestFixture.dbDir
+    val runtimeDir    = BlockStoreTestFixture.dbDir
+
+    implicit val logEff           = new LogStub[Task]
+    implicit val timeEff          = new LogicalTime[Task]
+    implicit val errorHandlerEff  = ApplicativeError_.applicativeError[Task, CommError](appErrTask)
+    implicit val nodeDiscoveryEff = new NodeDiscoveryStub[Task]()
+    implicit val transportLayerEff =
+      new TransportLayerTestImpl[Task](HashSetCasperTestNode.peerNode("taskNode", 40400),
+                                       Map.empty[PeerNode, mutable.Queue[Protocol]])
+    implicit val metricEff = new Metrics.MetricsNOP[Task]
+    implicit val blockStoreEff =
+      LMDBBlockStore.create[Task](LMDBBlockStore.Config(blockStoreDir, 1024L * 1024))(syncInstance,
+                                                                                      metricEff)
+    implicit val turanOracleEffect = SafetyOracle.turanOracle[Task]
+
+    val activeRuntime  = Runtime.create(runtimeDir, 1024L * 1024)
+    val runtimeManager = RuntimeManager.fromRuntime(activeRuntime)
+    val validatorId    = ValidatorIdentity(Ed25519.toPublic(sk), sk, "ed25519")
+
+    val casperTask = for {
+      _        <- blockStoreEff.put(genesis.blockHash, genesis)
+      blockMap <- blockStoreEff.asMap()
+    } yield
+      MultiParentCasper.hashSetCasper[Task](runtimeManager, Some(validatorId), genesis, blockMap)
+
+    def cleanUp(): Unit = {
+      activeRuntime.close()
+      blockStoreEff.close()
+      runtimeDir.recursivelyDelete()
+      blockStoreDir.recursivelyDelete()
+    }
+
+    (casperTask, cleanUp _)
+  }
+
+  private val appErrTask = new ApplicativeError[Task, CommError] {
+    def ap[A, B](ff: Task[A => B])(fa: Task[A]): Task[B] =
+      ApplicativeError[Task, Throwable].ap[A, B](ff)(fa)
+    def pure[A](x: A): Task[A] = ApplicativeError[Task, Throwable].pure[A](x)
+    def raiseError[A](e: CommError): Task[A] =
+      ApplicativeError[Task, Throwable].raiseError(new Exception(e.toString))
+
+    def handleErrorWith[A](fa: Task[A])(f: (CommError) => Task[A]): Task[A] = fa
+  }
+
+  private val syncInstance =
+    new Sync[Task] {
```

Comment:
I thought there's already an instance of `Sync[Task]` in monix.

Reply:
`CatsAsyncForTask`.

Reply:
It doesn't work though.
```
scala> import cats._
import cats._

scala> import monix.eval.Task
import monix.eval.Task

scala> import cats.effect._
import cats.effect._

scala> import cats.implicits._
import cats.implicits._

scala> val sync = implicitly[Sync[Task]]
sync: cats.effect.Sync[monix.eval.Task] = monix.eval.instances.CatsConcurrentForTask$@3658cc10

scala> val test = sync.bracketCase(Task.delay(0))(_ => Task.delay(1))((_, _) => Task.delay(()))
java.lang.AbstractMethodError: Method monix/eval/instances/CatsConcurrentForTask$.bracketCase(Ljava/lang/Object;Lscala/Function1;Lscala/Function2;)Ljava/lang/Object; is abstract
  at monix.eval.instances.CatsConcurrentForTask$.bracketCase(CatsAsyncForTask.scala)
  ... 36 elided
```

Reply:
What about https://github.com/monix/monix/blob/b9dc4a05ccafbdde4e0e2c0529748bc79bed1e89/monix-eval/shared/src/main/scala/monix/eval/instances/CatsAsyncForTask.scala#L56 ?

Reply:
I don't know what to tell you. You can see in the repl session that it resolves to `CatsConcurrentForTask` just as you linked to in the source code and then when I call `bracketCase` is says that the method is abstract.